### PR TITLE
fix: User gets an error even when table widget is added from the API …

### LINKED ISSUE #14037: [Bug]-[876]:User gets an error even when table widget is added from the API page successfully
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

When the table widget is added successfully from the API page we get to see the below error `The value at primaryColumns.action.buttonColor is invalid`. The API page is still adding the Table widget v1 hence this is a table widget v1 issue.

[LOOM DEMO](https://www.loom.com/share/9f3e514ffe3346ceab5098d61655fe96)

### Steps To Reproduce

1. Add a datasource and create a new API for the same.
2. After successfully running the the API and getting the proper response in the API page, in the `Add new widget` section of the same page click on the `Table` widget.

**Expected:** The table widget should be added to the canvas and no error should be thrown to the user.
**Actual:** The user sees `The value at primaryColumns.action.buttonColor is invalid` error.

### Public Sample App

_No response_

### Version

Cloud

### PR DESCRIPTION
**Pull request by Druthi**
## Description
fix: User gets an error even when table widget is added from the API page successfully

Fixes #14037 

## Type of change
- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?
- Manual

### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

### QA activity:
- [ ] Test plan has been approved by relevant developers
- [ ] Test plan has been peer reviewed by QA
- [x] Cypress test cases have been added and approved by either SDET or manual QA
- [ ] Organized project review call with relevant stakeholders after Round 1/2 of QA
- [x] Added Test Plan Approved label after reveiwing all Cypress test

### CODE REVIEW

File: app/client/src/sagas/WidgetAdditionSagas.ts [hunk 1]
```diff
@@ -138,7 +138,14 @@ function* getChildWidgetProps(
     widget,
     themeConfigWithoutChildStylesheet,
   );
-  widget.dynamicBindingPathList = clone(dynamicBindingPathList);
+
+  if (params.dynamicBindingPathList) {
```
Comment (ApekshaBhosale) [inline app/client/src/sagas/WidgetAdditionSagas.ts:142]: what is the value of dynamicBindingPathList in params? can you give example, please?

File: app/client/src/sagas/WidgetAdditionSagas.ts [hunk 2]
```diff
@@ -138,7 +138,14 @@ function* getChildWidgetProps(
...
+    const copy = [...dynamicBindingPathList, ...params.dynamicBindingPathList];
```
Comment (ApekshaBhosale) [inline app/client/src/sagas/WidgetAdditionSagas.ts:143]: please use a different variable name than `copy`

**Review by ApekshaBhosale — CHANGES_REQUESTED** [review]

**Review by ApekshaBhosale — APPROVED** [review]

**Comment by ramsaptami** [comment]
LGTM @Druthi

**Review by ApekshaBhosale — APPROVED** [review]

**Pull request merged**